### PR TITLE
Building and training the Neural Net

### LINKED ISSUE #2: [Feature 2] A neural network system
Develop a system that conducts forward propagation and backpropagation on a dataset.

### PR DESCRIPTION
**Pull request by Rishit-dagli**
This PR adds code to train the model.

Closes #2

### CODE REVIEW

File: src/main/java/com/playground/playground/modelling/NeuralNet.java
Comment (T3CHW1ZRD) [inline src/main/java/com/playground/playground/modelling/NeuralNet.java:1]: good neural network implementation

File: src/main/java/com/playground/playground/modelling/ModelTrainingServices.java
Comment (T3CHW1ZRD) [inline src/main/java/com/playground/playground/modelling/ModelTrainingServices.java:1]: looks good we may have to move the import dependencies to a higher layer later

File: build.gradle
Comment (T3CHW1ZRD) [inline build.gradle:1]: looks good

File: .github/CODEOWNERS
Comment (T3CHW1ZRD) [inline .github/CODEOWNERS:1]: yes

**Review by T3CHW1ZRD — APPROVED** [review]

**Review by Shivesh777 — COMMENTED** [review]
Some checks seem to fail, do you mind addressing them and modifying the code to fix that?

**Review by Shivesh777 — COMMENTED** [review]
I do not see where the training metrics are being stored, is that being upon in a different branch?

**Review by Ford-Robert — CHANGES_REQUESTED** [review]
Could you add some javadocs to the main classes, and most important methods to increase the readability of this code. Especially for the classes/functions that will be called by Use-case and Driver layers. Specifically when the play button is pressed which method is called to initiate the training process, and where are the weights stored for line-thinkness (if we end up implementing that feature)

**Review by Shivesh777 — DISMISSED** [review]

**Comment by Rishit-dagli** [comment]
> Could you add some javadocs to the main classes, and most important methods to increase the readability of this code

Ah, yes I need to do it for all methods, considering the time crunch we have and are yet to integrate this, I was thinking maybe I could do it but defer it to another PR as of now, what do you think?

> Specifically when the play button is pressed which method is called to initiate the training process, and where are the weights stored for line-thinkness (if we end up implementing that feature)

This is still WIP, working on those aspects yet, this is currently just the training code and nothing more.

**Comment by Rishit-dagli** [comment]
This PR is now ready for review

**Review by Ford-Robert — APPROVED** [review]

**Comment by Rishit-dagli** [comment]
/retitle Building and training the Neural Net

**Review by Shivesh777 — APPROVED** [review]

**Pull request merged**